code-challenges.test.js

```diff
@@ -22,9 +22,47 @@ const secretCodeWord2 = "Gobbledygook"
 const secretCodeWord3 = "Eccentric"
 // Expected output: "3cc3ntr1c"
 
+describe("secretCode", () => {
+    it("that takes in a string and returns a coded message", () => {
+        expect(secretCode(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+        expect(secretCode(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+        expect(secretCode(secretCodeWord3)).toEqual("3cc3ntr1c")
+    })
+})
 
 // b) Create the function that makes the test pass.
 
+// Pseudo Code:
+// Create a function expression that takes a string as a parameter. (secretCode)
+// Create a variable that invokes the split method on the string. (strArr)
+// Have the function invoke the forEach method on the variable.
+    // Nest if and else if statements into the forEach method, that will evaluate if the index value of the variable is equivalent to 'a', 'e', 'i' or 'o', respectively; and change the index value of the variable if the evaluation passes.
+// Create a variable that invokes the join method on the first variable. (theString)
+// Have the function return the second variable.
+// Run the function with the provided variables.
+
+let secretCode = (string) => {
+    strArr = string.split("");
+    strArr.forEach((value, index) => {
+        if(strArr[index].toLowerCase() == 'a') {
+            strArr[index] = 4
+        } else if (strArr[index].toLowerCase() == 'e') {
+            strArr[index] = 3
+        } else if (strArr[index].toLowerCase() == 'i') {
+            strArr[index] = 1
+        } else if (strArr[index].toLowerCase() == 'o') {
+            strArr[index] = 0
+        }
+    })
+    theString = strArr.join("")
+    return theString
+}
+
+console.log(secretCode(secretCodeWord1))
+
+console.log(secretCode(secretCodeWord2))
+
+console.log(secretCode(secretCodeWord3))
```

Comment:
Well done!  Just don't forget to use a variable declaration (let, const) in javascript  (line 57).